### PR TITLE
Added Shell Sort Algorithm in C++ (readme updated)

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Added Shell Sort in C language : 
Best complexity: n
Average complexity: n*log(n)^2 or n^(3/2)

#24 Change Requested.

### CODE REVIEW

File: shellSort.c [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+void shellSort(int array[], int n){
+  for (int gap = n/2; gap > 0; gap /= 2){
```
Comment (Bhavnaharitsa) [inline shellSort.c]: add program name, description and input output in the comment section

File: shellSort.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      }
+      array[j] = temp;
+    }
+  }
+}
+void printArray(int array[], int size){
+  for(int i=0; i<size; ++i){
+     printf("%d  ", array[i]);
+  }
+  printf("\n");
```
Comment (Bhavnaharitsa) [inline shellSort.c]: no need this

File: shellSort.c [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      int j;
+      for (j = i; j >= gap && array[j - gap] > temp; j -= gap){
+        array[j] = array[j - gap];
+      }
+      array[j] = temp;
+    }
+  }
+}
+void printArray(int array[], int size){
+  for(int i=0; i<size; ++i){
```
Comment (Bhavnaharitsa) [inline shellSort.c]: try outing int i=1; i<size-1; ++i if we are getting the same output

File: shellSort.c [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+     printf("%d  ", array[i]);
+  }
+  printf("\n");
+}
+int main(){
+  int data[]={9, 8, 3, 7, 5, 6, 4, 1};
```
Comment (Bhavnaharitsa) [inline shellSort.c]: make it as though you are getting data from user and not hardcoded

**Review by chanmol1999 — APPROVED** [review]

**Comment by chanmol1999** [comment]
Resolve the conflicts @Abhishek-kumar09

**Pull request merged**